File: leetcode/top150/189-rotate-array/python/solution.py
from templates.python.solution import SolutionBase

class Solution(SolutionBase):

    def __init__(self, *inputs):
        super().__init__(*inputs)

    def solution(self):
        for i in range(0, self.inputs[0]):
            # Rotate 1 time
            val = self.inputs[1][-1]
            self.inputs[1].pop(-1)
            self.inputs[1].insert(0, val)
        return

File: leetcode/top150/189-rotate-array/c++/solution.cpp
#include "templates/solution.hpp"

using namespace solution;

// Fill in this solution if a return value is expected
// Provide input/output types to function declaration
template<>
void Solution::solution(std::vector<std::shared_ptr<output<>>>& outputs,
                        std::vector<std::shared_ptr<input<int, std::vector<int>>>>& inputs) {
    return;
}

// Fill in this solution if no return value is expected
// Provide input types to function declaration
template<>
void Solution::inplace_solution(std::vector<std::shared_ptr<input<int, std::vector<int>>>>& inputs) {
    auto [steps, nums] = inputs[0]->inputs;

    for (size_t i = 0; i < steps; ++i) {
        // Rotate 1 step
        auto val = nums[nums.size() - 1];
        nums.pop_back();
        auto it = nums.begin();
        nums.insert(it, val);
    }
    
    std::get<1>(inputs[0]->inputs).swap(nums);
    return;
}